src/test/java/de/tum/in/www1/artemis/service/connectors/AtheneServiceTest.java

```diff
@@ -0,0 +1,265 @@
+package de.tum.in.www1.artemis.service.connectors;
+
+import static org.apache.commons.codec.digest.DigestUtils.sha1Hex;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.method;
+import static org.springframework.test.web.client.match.MockRestRequestMatchers.requestTo;
+import static org.springframework.test.web.client.response.MockRestResponseCreators.withSuccess;
+
+import java.time.ZonedDateTime;
+import java.util.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.client.ExpectedCount;
+import org.springframework.test.web.client.MockRestServiceServer;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationBambooBitbucketJiraTest;
+import de.tum.in.www1.artemis.domain.*;
+import de.tum.in.www1.artemis.domain.enumeration.Language;
+import de.tum.in.www1.artemis.repository.*;
+import de.tum.in.www1.artemis.service.TextAssessmentQueueService;
+import de.tum.in.www1.artemis.service.TextBlockService;
+import de.tum.in.www1.artemis.service.TextSubmissionService;
+import de.tum.in.www1.artemis.util.ModelFactory;
+import de.tum.in.www1.artemis.web.rest.dto.AtheneDTO;
+
+public class AtheneServiceTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    TextBlockService textBlockService;
+
+    @Autowired
+    TextAssessmentQueueService textAssessmentQueueService;
+
+    @Mock
+    TextExerciseRepository textExerciseRepository;
+
+    @Mock
+    TextBlockRepository textBlockRepository;
+
+    @Mock
+    TextSubmissionService textSubmissionService;
+
+    @Mock
+    TextClusterRepository textClusterRepository;
+
+    private final String API_ENDPOINT = "http://localhost/submit";
+
+    AtheneService atheneService;
+
+    TextExercise exercise1;
+
+    /**
+     * Initializes atheneService and example exercise
+     */
+    @BeforeEach
+    public void init() {
+        // Create atheneService and inject @Value fields
+        atheneService = new AtheneService(textSubmissionService, textBlockRepository, textBlockService, textClusterRepository, textExerciseRepository, textAssessmentQueueService);
+        ReflectionTestUtils.setField(atheneService, "ARTEMIS_SERVER_URL", ARTEMIS_SERVER_URL);
```

Comment:
I consider ReflectionTestUtils.setField bad practice here.
Instead let the `application-artemis.yml` in the test folder set those values for test cases automatically